ethcore/sync/src/chain/mod.rs

```diff
@@ -1351,6 +1367,19 @@ impl ChainSync {
 	pub fn propagate_private_transaction(&mut self, io: &mut SyncIo, transaction_hash: H256, packet_id: SyncPacket, packet: Bytes) {
 		SyncPropagator::propagate_private_transaction(self, io, transaction_hash, packet_id, packet);
 	}
+
+	/// Request private state from peers
+	pub fn request_private_state(&mut self, io: &mut SyncIo, hash: &H256) {
+		let private_state_peers = self.get_private_state_peers();
+		if private_state_peers.is_empty() {
+			error!(target: "privatetx", "Cannot request private state, no peers with private tx enabled connected");
+		} else {
+			trace!(target: "privatetx", "Requesting private stats from {:?}", private_state_peers);
+			for peer_id in private_state_peers {
```

Comment:
> Is this a good idea to ask all available peers for the same data?

In general case, no. But from practical standpoint for private transactions it is more than ok. From testing and other issues I've seen, that usually it's not so many peers with private transactions functionality enabled around. And it's better to ask more, than not to receive anything.

Reply:
> Also you could ask for multiple states at once from `request_private_state()` => this could trigger multiple warnings from [here](https://github.com/paritytech/parity-ethereum/blob/master/ethcore/sync/src/chain/requester.rs#L141) because you do not check `peer.asking` anywhere.

Also good one. Will rework.